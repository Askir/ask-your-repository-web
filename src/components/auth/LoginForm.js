// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { flashErrorMessage } from 'redux-flash';
import GoogleLogin from 'react-google-login';
import Form from 'react-validation/build/form';
import Input from 'react-validation/build/input';
import Button from 'react-validation/build/button';
import { login, loginWithGoogle } from '../../state/auth/auth.actionCreators';
import './LoginForm.scss';

type Props = {
  dispatch: Function,
};

type State = {
  email: string,
  password: string,
};

class LoginForm extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = (e) => {
    e.preventDefault();

    const { email, password } = this.state;
    const { dispatch } = this.props;
    if (email && password) {
      dispatch(login(email, password));
    }
  };

  handleGoogleSuccess = (googleUser) => {
    this.props.dispatch(loginWithGoogle(googleUser));
  };

  handleGoogleFailure = () => {
    const { dispatch } = this.props;
    dispatch(flashErrorMessage('Google login failed'));
  };

  render() {
    const { email, password } = this.state;
    return (
      <Form onSubmit={this.handleSubmit} className="LoginForm">
        <h1>Login</h1>
        <div className="form-input">
          <label>
            Email or Username:
            <Input
              type="text"
              name="email"
              value={email}
              onChange={this.handleChange}
              data-cy="login-email-input"
            />
          </label>
        </div>
        <div className="form-input">
          <label>
            Password:
            <Input
              type="password"
              name="password"
              value={password}
              onChange={this.handleChange}
              data-cy="login-password-input"
            />
          </label>
        </div>
        <div>
          <Link to="/register">No account yet? Register here.</Link>
        </div>
        <div className="LoginForm__buttons">
          <Button data-cy="login-submit-button">
            Submit
          </Button>
          <GoogleLogin
            clientId="24456970850-u9prmb53vbgthab5qrel3b0rqe7rcjua.apps.googleusercontent.com"
            buttonText="Sign up with Google"
            className="google-login"
            onSuccess={this.handleGoogleSuccess}
            onFailure={this.handleGoogleFailure}
          />
        </div>
      </Form>
    );
  }
}

export default connect()(LoginForm);
