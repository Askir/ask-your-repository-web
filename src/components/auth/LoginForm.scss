.LoginForm {
  margin: 40px auto;
  width: 80%;
  max-width: 540px;
  padding: 0 5%;

  h1 {
    font-weight: bold;
  }

  a {
    color: lightslategray;
  }

  &__buttons {
    display: flex;
    margin-top: 16px;

    button {
      display: inline-block;
      background-color: green;
      width: 48%;
      font-weight: bold;
      padding: 1em 2em;
      border-radius: .3em;
    }

    .google-login {
      background-color: gray;
      display: inline-block;
      color: white;
      margin-left: 4%;
      width: 48%;
      border-radius: .3em !important;
      text-align: center;

      div {
        padding-left: 15px !important;
      }
    }
  }
}
