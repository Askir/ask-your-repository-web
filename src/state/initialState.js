// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as image } from './image/image.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  image,
  activeTeam,
  teamSidebar,
};

export default initialState;
