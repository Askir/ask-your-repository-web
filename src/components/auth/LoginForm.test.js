// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import LoginForm from './LoginForm';
import initialState from '../../state/initialState';

const mockStore = configureMockStore();

jest.mock('../../services/AuthService', () => ({
  logout: jest.fn(() => Promise.resolve()),
}));


describe('<LoginForm />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount((
      <Provider store={store}>
        <MemoryRouter>
          <LoginForm />
        </MemoryRouter>
      </Provider>
    )).find('LoginForm');
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
