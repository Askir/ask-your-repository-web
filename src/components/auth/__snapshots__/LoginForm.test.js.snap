// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<LoginForm
  dispatch={[Function]}
>
  <Form(t)
    className="Form"
    onSubmit={[Function]}
  >
    <t
      className="Form"
      getValues={[Function]}
      hideError={[Function]}
      onSubmit={[Function]}
      showError={[Function]}
      validate={[Function]}
      validateAll={[Function]}
    >
      <form
        className="Form"
        onSubmit={[Function]}
      >
        <div
          className="Form__title"
        >
          Login
        </div>
        <div
          className="Form__external-login"
        >
          <Connect(GoogleLogin)>
            <GoogleLogin
              dispatch={[Function]}
            >
              <e
                accessType="online"
                buttonText="Sign in with Google"
                className="Form__external-login__google"
                cookiePolicy="single_host_origin"
                disabledStyle={
                  Object {
                    "opacity": 0.6,
                  }
                }
                fetchBasicProfile={true}
                icon={true}
                isSignedIn={false}
                jsSrc="https://apis.google.com/js/api.js"
                onFailure={[Function]}
                onRequest={[Function]}
                onSuccess={[Function]}
                prompt=""
                scope="profile email"
                tag="button"
                theme="dark"
                type="button"
                uxMode="popup"
              >
                <button
                  className="Form__external-login__google"
                  disabled={true}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgb(66, 133, 244)",
                      "border": "1px solid transparent",
                      "borderRadius": 2,
                      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                      "color": "#fff",
                      "display": "inline-flex",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "opacity": 0.6,
                      "padding": 0,
                    }
                  }
                  type="button"
                >
                  <Component
                    active={false}
                    key="1"
                  >
                    <div
                      style={
                        Object {
                          "background": "#fff",
                          "borderRadius": 2,
                          "marginRight": 10,
                          "padding": 10,
                        }
                      }
                    >
                      <svg
                        height="18"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="#000"
                          fillRule="evenodd"
                        >
                          <path
                            d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                            fill="#EA4335"
                          />
                          <path
                            d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                            fill="#4285F4"
                          />
                          <path
                            d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                            fill="#FBBC05"
                          />
                          <path
                            d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                            fill="#34A853"
                          />
                          <path
                            d="M0 0h18v18H0z"
                            fill="none"
                          />
                        </g>
                      </svg>
                    </div>
                  </Component>
                  <b
                    icon={true}
                    key="2"
                  >
                    <span
                      style={
                        Object {
                          "fontWeight": 500,
                          "paddingBottom": 10,
                          "paddingLeft": 0,
                          "paddingRight": 10,
                          "paddingTop": 10,
                        }
                      }
                    >
                      Sign in with Google
                    </span>
                  </b>
                </button>
              </e>
            </GoogleLogin>
          </Connect(GoogleLogin)>
        </div>
        <hr
          className="Form__separator"
          data-content="OR"
        />
        <div
          className="form-input"
        >
          <label
            className="Form__label"
          >
            Email or Username:
            <Control(s)
              data-cy="login-email-input"
              name="email"
              onChange={[Function]}
              type="text"
              validations={Array []}
              value=""
            >
              <s
                data-cy="login-email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              >
                <div>
                  <input
                    data-cy="login-email-input"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div
          className="form-input"
        >
          <label
            className="Form__label"
          >
            Password:
            <Link
              className="Form__label-link"
              replace={false}
              to="/forgot-password"
            >
              <a
                className="Form__label-link"
                href="/forgot-password"
                onClick={[Function]}
              >
                Forgot password?
              </a>
            </Link>
            <Control(s)
              data-cy="login-password-input"
              name="password"
              onChange={[Function]}
              type="password"
              validations={Array []}
              value=""
            >
              <s
                data-cy="login-password-input"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                type="password"
                value=""
              >
                <div>
                  <input
                    data-cy="login-password-input"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div>
          <Link
            className="Form__link"
            replace={false}
            to="/register"
          >
            <a
              className="Form__link"
              href="/register"
              onClick={[Function]}
            >
              No account yet? Register here.
            </a>
          </Link>
        </div>
        <div
          className="Form__buttons"
        >
          <Button(s)
            className="Form__buttons__item"
            data-cy="login-submit-button"
          >
            <s
              className="Form__buttons__item"
              data-cy="login-submit-button"
              hasErrors={false}
            >
              <button
                className="Form__buttons__item"
                data-cy="login-submit-button"
                disabled={false}
              >
                Login
              </button>
            </s>
          </Button(s)>
          <Link
            className="Form__buttons__cancel"
            replace={false}
            to="/"
          >
            <a
              className="Form__buttons__cancel"
              href="/"
              onClick={[Function]}
            >
              Cancel
            </a>
          </Link>
        </div>
      </form>
    </t>
  </Form(t)>
</LoginForm>
`;
